packages/main/src/SplitButton.js

```diff
@@ -0,0 +1,449 @@
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.SplitButton
+ * @extends UI5Element
+ * @tagname ui5-split-button
+ * @public
+ * @since 1.1.0
+ */
+class SplitButton extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return SplitButtonCss;
+	}
+
+	static get template() {
+		return SplitButtonTemplate;
+	}
+
+	static async onDefine() {
+		SplitButton.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	onAfterRendering() {
+		this._textButtonIcon = this.buttonIcon;
+	}
+
+	_onFocusOut(event) {
+		if (this.disabled || event.isMarked) {
+			return;
+		}
+
+		this._focusedState(false);
+	}
+
+	_onFocusIn(event) {
+		if (this.disabled || event.isMarked) {
+			return;
+		}
+
+		this._focusedState(true);
+	}
+
+	_onMouseDownArrow(event) {
+		this._focusedState(false);
+	}
+
+	_onMouseTouchDownText(event) {
+		this._focusedState(false);
+		this._textButtonActiveState(true);
+	}
+
+	_onMouseUpText() {
+		this._textButtonActiveState(false);
+	}
+
+	_onKeyDown(event) {
+		if (isTabNext(event) || isTabPrevious(event)) {
+			this._tabIndexUpdate = true;
+		} else if (isDown(event) || isUp(event) || isDownAlt(event) || isUpAlt(event) || isF4(event)) {
+			this._arrowButtonActive = true;
+			this._keyboardAction = true;
+			this._fireArrowClick();
+		} else if (isSpace(event) || isEnter(event)) {
+			this._arrowButtonActive = false;
+			this._updateIcon = true;
+			this._textButtonActiveState(true);
+			if (isEnter(event)) {
+				this._keyboardAction = true;
+				event.preventDefault();
+				this._fireClick();
+			} else {
+				this._keyboardAction = true;
+			}
+		}
+
+		if (this._keyboardAction && (isEscape(event) || isShift(event))) {
+			this._shiftOrEscapePressed = true;
+			this._textButtonActiveState(false);
+		}
+	}
+
+	_onKeyUp(event) {
+		if (isDown(event) || isUp(event) || isDownAlt(event) || isUpAlt(event) || isF4(event)) {
+			this._arrowButtonActive = false;
+		} else if (isSpace(event) || isEnter(event)) {
+			this._textButtonActiveState(false);
+			if (isSpace(event)) {
+				if (!this._shiftOrEscapePressed) {
+					event.preventDefault();
+					event.stopPropagation();
+					this._fireClick();
+				} else {
+					this._shiftOrEscapePressed = false;
+				}
+			}
+		}
+	}
+
+	_focusedState(state) {
+		this._tabIndexUpdate = true;
+		this.focused = state;
+	}
+
+	_textButtonActiveState(state) {
+		this._textButtonActive = state;
+		this._textButtonIcon = this.buttonIcon;
+	}
+
+	_fireClick(event) {
+		event && event.stopPropagation();
+		this.fireEvent("click");
+	}
+
+	_fireArrowClick(event) {
+		event && event.stopPropagation();
+		this.fireEvent("arrow-click");
+	}
+
+	get buttonIcon() {
+		return this.textButton && this.activeIcon !== "" && this.textButton.active ? this.activeIcon : this.icon;
+	}
+
+	get textButton() {
+		return this.shadowRoot.querySelector(".ui5-split-text-button");
```

Comment:
Please use this.getDomRef().querySelector

Reply:
Done.